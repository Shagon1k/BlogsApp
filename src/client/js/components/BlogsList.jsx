import React from 'react';
import Blog from './Blog.jsx';

const BlogsList = ({blogs, onBlogDelete}) => (
	<ul className="blogList">
		{
			blogs.map((i) => <Blog key={i.id} blog={i.blog} onDelete={() => onBlogDelete(i.id, i.blog)} />)
		}
	</ul>
)

export default BlogsList;