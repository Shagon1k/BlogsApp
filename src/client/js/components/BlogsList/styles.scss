.blogList {
	display: flex;
	flex-flow: column wrap;
	align-items: center;
	justify-content: center;

	.blogBox {
		position: relative;
		box-sizing: border-box;
		width: 100%;
		background: #ccd2dd;
		border-radius: 5px;
		padding: 15px;
		margin-bottom: 10px;

		.blogInfo {
			display: flex;
			flex-flow: row wrap;
			align-items: flex-start;

			dt {
				flex-basis: 20%;
				margin-bottom: 5px;
				font-weight: bold;
			}

			dd {
				flex-basis: 80%;
				margin-bottom: 5px;
				text-align: justify;
			}
		}
		
		.removeBlog {
			position: absolute;
			top: 12px;
			right: 12px;
			font-size: 20px;
			font-weight: bold;
			color: #961e1e;
		}
	}
}