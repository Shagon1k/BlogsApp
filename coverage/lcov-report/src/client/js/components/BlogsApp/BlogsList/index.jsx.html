<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for src/client/js/components/BlogsApp/BlogsList/index.jsx</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="..\..\..\..\..\..\prettify.css" />
    <link rel="stylesheet" href="..\..\..\..\..\..\base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(..\..\..\..\..\..\sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="..\..\..\..\..\..\index.html">All files</a> / <a href="index.html">src/client/js/components/BlogsApp/BlogsList</a> index.jsx
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">83.78% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>31/37</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">54.84% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>17/31</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>12/12</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>9/9</span>
      </div>
    </div>
  </div>
  <div class='status-line high'></div>
<pre><table class="coverage">
<tr><td class="line-count quiet">1
2
3
4
5
6
7
8
9
10
11
12
13
14
15
16
17
18
19
20
21
22
23
24
25
26</td><td class="line-coverage quiet"><span class="cline-any cline-yes">4x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">2x</span></td><td class="text"><pre class="prettyprint lang-js">import React from 'react';<span class="cstat-no" title="statement not covered" ><span class="missing-if-branch" title="if path not taken" >I</span><span class="missing-if-branch" title="else path not taken" >E</span><span class="missing-if-branch" title="else path not taken" >E</span></span>
import Blog from '../Blog/index.jsx';
import './styles.scss';<span class="cstat-no" title="statement not covered" ><span class="cstat-no" title="statement not covered" ><span class="cstat-no" title="statement not covered" ><span class="branch-1 cbranch-no" title="branch not covered" ><span class="missing-if-branch" title="else path not taken" >E</span><span class="branch-1 cbranch-no" title="branch not covered" ><span class="missing-if-branch" title="if path not taken" >I</span><span class="branch-2 cbranch-no" title="branch not covered" ><span class="branch-1 cbranch-no" title="branch not covered" ><span class="branch-0 cbranch-no" title="branch not covered" ><span class="missing-if-branch" title="if path not taken" >I</span><span class="missing-if-branch" title="if path not taken" >I</span></span></span></span></span></span></span></span></span>
&nbsp;
&nbsp;
class BlogsList extends React.Component {
	constructor(props) {<span class="branch-1 cbranch-no" title="branch not covered" ></span>
		super(props);
	}
&nbsp;
	componentDidMount() {
		this.props.initBlogs();
	}
&nbsp;
	render() {
		return (
			&lt;ul className="blogList"&gt;
				{
					this.props.blogs.map((i) =&gt; &lt;Blog key={i.id} blog={i.blog} blogId={i.id} /&gt;)
				}
			&lt;/ul&gt;
		);
	}
}
&nbsp;
export default BlogsList;<span class="cstat-no" title="statement not covered" ><span class="cstat-no" title="statement not covered" ><span class="missing-if-branch" title="else path not taken" >E</span></span></span></pre></td></tr>
</table></pre>
<div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Sat Mar 03 2018 15:03:49 GMT+0300 (Belarus Standard Time)
</div>
</div>
<script src="..\..\..\..\..\..\prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="..\..\..\..\..\..\sorter.js"></script>
</body>
</html>
