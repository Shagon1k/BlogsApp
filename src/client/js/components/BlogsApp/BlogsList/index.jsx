import React from 'react';
import Blog from '../Blog/index.jsx';
import './styles.scss';


class BlogsList extends React.Component {
	constructor(props) {
		super(props);
	}

	componentDidMount() {
		this.props.initBlogs();
	}

	render() {
		return (
			<ul className="blogList">
				{
					this.props.blogs.map((i) => <Blog key={i.id} blog={i.blog} blogId={i.id} />)
				}
			</ul>
		);
	}
}

export default BlogsList;