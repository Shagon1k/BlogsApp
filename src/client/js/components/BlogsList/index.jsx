import React from 'react';
import Blog from '../Blog/index.jsx';
import './styles.scss';

const BlogsList = ({blogs, onBlogDelete}) => (
	<ul className="blogList">
		{
			blogs.map((i) => <Blog key={i.id} blog={i.blog} onDelete={() => onBlogDelete(i.id, i.blog)} />)
		}
	</ul>
)

export default BlogsList;