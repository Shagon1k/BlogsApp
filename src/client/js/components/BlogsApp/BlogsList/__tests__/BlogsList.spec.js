import React from 'react';
import BlogsList from '../index.jsx';

describe('BlogsList component', () => {
	const blogs = [
		{
			blog: {},
			id: 1
		},
		{
			blog: {},
			id: 2
		},
		{
			blog: {},
			id: 3
		}
	];
	const initBlogs = jest.fn();

	it('renders correctly', () => {
		const wrapper = shallow(<BlogsList blogs={blogs} initBlogs={initBlogs}/>);
		expect(wrapper).toMatchSnapshot();
		expect(initBlogs).toBeCalled();

	});
})