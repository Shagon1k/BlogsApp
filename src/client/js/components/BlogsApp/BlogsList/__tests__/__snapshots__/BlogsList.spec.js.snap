// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BlogsList component renders correctly 1`] = `
<ul
  className="blogList"
>
  <Blog
    blog={Object {}}
    blogId={1}
    key="1"
  />
  <Blog
    blog={Object {}}
    blogId={2}
    key="2"
  />
  <Blog
    blog={Object {}}
    blogId={3}
    key="3"
  />
</ul>
`;
